geomesa-feature/src/test/scala/org/locationtech/geomesa/feature/SerializationTester.scala

```diff
@@ -57,35 +58,63 @@ object SerializationTester {
         sf.setAttribute("f6", new SimpleDateFormat("yyyyMMdd").parse("20140102"))
         sf.setAttribute("f7", GeohashUtils.wkt2geom("POINT(45.0 49.0)").asInstanceOf[Point])
         sf.setAttribute("f8", GeohashUtils.wkt2geom("POLYGON((-80 30,-80 23,-70 30,-70 40,-80 40,-80 30))").asInstanceOf[Polygon])
+        sf.setAttribute("f9", List(r.nextDouble(), r.nextDouble(), r.nextDouble(), r.nextDouble(), r.nextDouble()))
+        sf.setAttribute("f10", Map(r.nextString(10) -> r.nextInt(),
+          r.nextString(10) -> r.nextInt(), r.nextString(10) -> r.nextInt(), r.nextString(10) -> r.nextInt()))
         list += sf
       }
       list.toList
     }
 
-    val features = createComplicatedFeatures(500000)
+//    size avro: 1151750
+//    size kryo: 1137102 1% smaller
 
-    def one() = {
-      val oldBaos = new ByteArrayOutputStream()
-      features.foreach { f =>
-        oldBaos.reset()
-        f.write(oldBaos)
-        oldBaos.toByteArray
-      }
-    }
+//    encode avro: 6.565
+//    encode kryo: 4.905 34% faster!
+
+//    decode avro: 9.455
+//    decode kryo: 7.625 24% faster!
+
+//    encode/decode avro: 16.02
+//    encode/decode kryo: 12.53 28% faster!
+
+//    complex attributes size avro: 1301239
+//    complex attributes size kryo: 1111239 17% smaller
+
+//    complex attributes encode/decode avro: 30.980
+//    complex attributes encode/decode kryo: 17.326 79% faster!
```

Comment:
i think one day i'll do the whole compiled avro vs thrift vs kryo thing too. see what pops out.